services/src/northbound/src/main/java/org/openkilda/northbound/controller/LinkController.java

```diff
@@ -113,4 +115,21 @@
     public CompletableFuture<BatchResults> delLinkProps(@RequestBody List<LinkPropsDto> keysAndProps) {
         return linkService.delLinkProps(keysAndProps);
     }
+
+    /**
+     * Get all flows for a particular link.
+     *
+     * @return list of flows for a particular link.
+     */
+    @ApiOperation(value = "Get all flows for a particular link, based on arguments.", response = FlowPayload.class,
+            responseContainer = "List")
+    @GetMapping(path = "/links/flows",
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public CompletableFuture<List<FlowPayload>> getFlowsForLink(@RequestParam(value = "src_switch") SwitchId srcSwitch,
```

Comment:
I still think that use query string arguments for mandatory ... arguments are bad idea.